src/main/java/org/prebid/server/bidder/readpeak/ReadPeakBidder.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            final List<BidderBid> bids = extractBids(bidResponse);
+            return Result.withValues(bids);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Empty SeatBid array");
+        }
+        return bidsFromResponse(bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(List::stream)
+                .map(bid -> {
+                    final Bid resolvedBid = resolveMacros(bid);
+                    final BidType bidType = getBidType(bid);
+                    final Bid updatedBid = addBidMeta(resolvedBid);
+                    return BidderBid.of(updatedBid, bidType, bidResponse.getCur());
+                })
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private static Bid resolveMacros(Bid bid) {
+        final BigDecimal price = bid.getPrice();
+        final String priceAsString = price != null ? price.toPlainString() : "0";
+
+        return bid.toBuilder()
+                .nurl(StringUtils.replace(bid.getNurl(), PRICE_MACRO, priceAsString))
+                .adm(StringUtils.replace(bid.getAdm(), PRICE_MACRO, priceAsString))
+                .build();
+    }
+
+    private static BidType getBidType(Bid bid) {
+        final Integer markupType = bid.getMtype();
+        if (markupType == null) {
+            throw new PreBidException("Missing MType for bid: " + bid.getId());
+        }
+
+        return switch (markupType) {
+            case 1 -> BidType.banner;
+            case 4 -> BidType.xNative;
+            default -> throw new PreBidException(
+                    "Unable to fetch mediaType " + bid.getMtype() + " in multi-format: " + bid.getImpid());
+        };
+    }
+
+    private Bid addBidMeta(Bid bid) {
+        final ObjectNode bidExt = bid.getExt() != null
+                ? bid.getExt().deepCopy()
+                : mapper.mapper().createObjectNode();
+
+        final ObjectNode prebidNode = bidExt.putObject("prebid");
+        final ObjectNode metaNode = prebidNode.putObject("meta");
+        metaNode.set("advertiserDomains", mapper.mapper().valueToTree(bid.getAdomain()));
+        return bid.toBuilder()
+                .ext(bidExt)
+                .build();
```

Comment:
I believe you can use this class `org.prebid.server.proto.openrtb.ext.response.ExtBidPrebidMeta`